tests/test_pay.py

```diff
@@ -2620,16 +2620,13 @@ def test_partial_payment(node_factory, bitcoind, executor):
     l2.daemon.wait_for_log(r'dev_disconnect')
     l3.daemon.wait_for_log(r'dev_disconnect')
 
-    # Now continue, payments will fail because receiver doesn't do MPP.
+    # Now continue, payments will succeed due to MPP.
     l2.rpc.dev_reenable_commit(l4.info['id'])
     l3.rpc.dev_reenable_commit(l4.info['id'])
 
-    # See FIXME in peer_htlcs: WIRE_INVALID_ONION_PAYLOAD would be better.
-    with pytest.raises(RpcError, match=r'WIRE_FINAL_INCORRECT_HTLC_AMOUNT'):
-        l1.rpc.call('waitsendpay', [inv['payment_hash'], None, 1])
-    with pytest.raises(RpcError, match=r'WIRE_FINAL_INCORRECT_HTLC_AMOUNT'):
-        l1.rpc.call('waitsendpay', {'payment_hash': inv['payment_hash'],
-                                    'parallel_id': 2})
+    l1.rpc.call('waitsendpay', [inv['payment_hash'], None, 1])
+    l1.rpc.call('waitsendpay', {'payment_hash': inv['payment_hash'],
+                                'parallel_id': 2})
```

Comment:
These need checks that we actually get the parts in the correct call, and that there isn't a place we cross wires :-)